ang/civicase/ContactCaseTabCaseDetails.js

```diff
@@ -0,0 +1,29 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  module.directive('civicaseContactCaseTabCaseDetails', function () {
+    return {
+      restrict: 'EA',
+      replace: true,
+      controller: ContactCaseTabCaseDetailsController,
+      templateUrl: '~/civicase/ContactCaseTabCaseDetails.html',
+      scope: {
+        item: '=selectedCase'
+      }
+    };
+  });
+
+  function ContactCaseTabCaseDetailsController ($scope) {
+    $scope.getContactRole = function (caseObj) {
```

Comment:
Doc missing

Reply:
This function is already inside ContactCaseTab. Lets not duplicate logic. Either pass the role from upper component to this one, or move the common function to a service. I would prefer 1st one.